packages/tools-iottwinmaker/src/lib/workspace.ts

```diff
@@ -318,5 +326,142 @@ async function createWorkspaceIfNotExists(workspaceId: string) {
+  if (instanceProfileResp.InstanceProfiles != undefined) {
+    for (const instanceProfile of instanceProfileResp.InstanceProfiles) {
+      await aws().iam.removeRoleFromInstanceProfile({
+        RoleName: roleName,
+        InstanceProfileName: instanceProfile.InstanceProfileName,
+      });
+      console.log(`removed role from instance profile: ${instanceProfile.InstanceProfileName}`);
+    }
+  }
+  // Step 2: Delete all policies that are associated with the role
+  const rolePolicies = await aws().iam.listRolePolicies({ RoleName: roleName });
+  if (rolePolicies.PolicyNames != undefined) {
+    for (const policy of rolePolicies.PolicyNames) {
+      await aws().iam.deleteRolePolicy({ RoleName: roleName, PolicyName: policy });
+      console.log(`deleted policy: ${policy}`);
+    }
+  }
+  const attachedRolePolicies = await aws().iam.listAttachedRolePolicies({ RoleName: roleName });
+  if (attachedRolePolicies.AttachedPolicies != undefined) {
+    for (const attachedPolicy of attachedRolePolicies.AttachedPolicies) {
+      await aws().iam.detachRolePolicy({ RoleName: roleName, PolicyArn: attachedPolicy.PolicyArn });
+      await aws().iam.deletePolicy({ PolicyArn: attachedPolicy.PolicyArn });
+      console.log(`detached and deleted policy: ${attachedPolicy.PolicyName}`);
+    }
+  }
+  // Step 3: Delete role
+  await aws().iam.deleteRole({ RoleName: roleName });
+  console.log(`deleted role: ${roleName}`);
+}
+
+/**
+ * Deletes a specified TwinMaker workspace
+ * @param workspaceId TM workspaceID
+ * @returns DeleteWorkspaceCommandOutput if successful
+ */
+async function deleteWorkspace(workspaceId: string): Promise<DeleteWorkspaceCommandOutput | undefined> {
+  try {
+    const deleteWorkspaceResp: DeleteWorkspaceCommandOutput = await aws().tm.deleteWorkspace({ workspaceId });
+    console.log(`deleted workspace: ${workspaceId}`);
+    return deleteWorkspaceResp;
+  } catch (err) {
+    console.log(`Unable to delete workspace ${workspaceId}:\n ${err}`);
+  }
+}
+
+/**
+ * Helper function that deletes all objects in a specified bucket, any potential object versions, and the bucket itself
+ * @param s3BucketName S3 bucket name
+ */
+async function deleteS3Bucket(s3BucketName: string | undefined) {
+  let isTruncated: boolean | undefined = true;
+  let NextContinuationToken;
+  // Delete all objects in bucket
+  while (isTruncated) {
+    const listObjectsResp: ListObjectsV2CommandOutput = await aws().s3.listObjectsV2({
+      Bucket: s3BucketName,
+      ContinuationToken: NextContinuationToken,
+    });
+    isTruncated = listObjectsResp.IsTruncated;
+    NextContinuationToken = listObjectsResp.NextContinuationToken;
+    if (listObjectsResp.Contents != undefined) {
+      for (const obj of listObjectsResp.Contents) {
+        await aws().s3.deleteObject({ Bucket: s3BucketName, Key: obj['Key'] });
+        console.log(`deleted S3 Object: ${obj['Key']}`);
+      }
+    }
+  }
+  // Delete all versions in bucket if bucket has versioning
```

Comment:
Just want to confirm: for AWS CLI there are high-level commands to delete the whole bucket (e.g. with `rm` [[1]](https://docs.aws.amazon.com/cli/latest/userguide/cli-services-s3-commands.html)) I'm guessing they're not available with the SDK?

Reply:
Correct as far as I have seen. Following rough steps outlined here: https://docs.aws.amazon.com/AmazonS3/latest/userguide/delete-bucket.html
(this version is in Java though but same process w/ SDK)